google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -65,15 +66,17 @@
       "https://cloud.google.com/sdk/docs/"
           + "#install_the_latest_cloud_tools_version_cloudsdk_current_version";
 
+  private boolean settingsModified;
```

Comment:
I wish there was a better way to do this. I remember we discussed something previously where there was a way to make dialogs aware of changes to certain swing components that it wasn't registering. But I do see that its complicated by the fact that this is just a panel that is included in various contexts. Do you know if its at all possible to update this?

Reply:
You're right, I think IDE uses something for `SettingsEditor` to track standard components state, but this is abstract `Configurable` with `createComponent()` and I think it's not tracked by default. Looks pretty ugly.

Reply:
It's called `UserActivityWatcher`, it looks like it's working pretty good with standard components we have here in panel, and simplified code a bit. We still use the flag to mark the state changes between `apply()`/`reset()` but it's easier to read now.